packages/grafana-data/src/types/datasource.ts

```diff
@@ -563,6 +563,7 @@ export interface DataSourceSettings<T extends DataSourceJsonData = DataSourceJso
   readOnly: boolean;
   withCredentials: boolean;
   version?: number;
+  accessControl?: KeyValue<boolean>;
```

Comment:
Would it be a good idea to type this one to something like: `DataSourceAccessControl`?

Reply:
Or will it contain more scopes than just the ones for the data source?

Reply:
Yes, the idea would be to cover more types, for instance `Organization` and `Users`. This is why I defined the `ObjectWithMetadata` type in public/app/types/accessControl.ts. However, given your comment on the type, I assume I should move its definition to grafana-data?

Reply:
Doesn't it feel a bit weird to have other scopes than for the current datasource added to the datasource?

Reply:
Oh my apologies, I misunderstood your questions :pray:. You are right, only actions related to the current data source will be listed here (ex: `datasources:read`, `datasources:delete`)
However `DataSourceSettings` won't be the only interface with `accessControl` metadata. Other interfaces such as `Organization` and `UserDTO` will have the same `accessControl` field, hence the generic name :slightly_smiling_face: